zerver/actions/realm_settings.py

```diff
@@ -186,13 +186,30 @@ def do_set_realm_message_editing(
     send_event(realm, event, active_user_ids(realm.id))
 
 
-def do_set_realm_notifications_stream(
-    realm: Realm, stream: Optional[Stream], stream_id: int, *, acting_user: Optional[UserProfile]
+def do_set_realm_stream(
```

Comment:
This refactor is probably a good idea -- this whole code path is kinda duplicated 3x, but this bit is probably the most complex. 

I would prefer to have it in a preparatory commit (you could even split that out as a separate prep PR), since it's one of the few changes in this PR that adjusts existing code.

Reply:
DONE. PR #22263